kibom/component.py

```diff
@@ -330,19 +330,23 @@ def isFitted(self):
 
         # Variants logic
         opts = check.split(",")
+        # Only fit for ...
+        exclusive = False
         for opt in opts:
             opt = opt.strip()
             # Any option containing a DNF is not fitted
             if opt in DNF:
                 return False
             # Options that start with '-' are explicitly removed from certain configurations
-            if opt.startswith("-") and str(opt[1:]) in self.prefs.pcbConfig:
+            if opt.startswith("-") and opt[1:] in self.prefs.pcbConfig:
                 return False
             # Options that start with '+' are fitted only for certain configurations
-            if opt.startswith("+") and opt[1:] not in self.prefs.pcbConfig:
-                return False
-
-        return True
+            if opt.startswith("+"):
+                exclusive = True
+                if opt[1:] in self.prefs.pcbConfig:
+                    return True
```

Comment:
I think this contradicts documented behavior:
> 
    If the fit_field contains the variant beginning with a '-' character, it will be excluded irrespective of any other '+' matches.
    If the fit_field contains the variant beginning with a '+' and matches any of the given variants, it will be included.

And will behave incorrectly e.g. for `fit_field='+A,-B'` when running with `pcbConfig=['A','B']`.

Please also make sure that behavior of your code is invariant to the order of options in `opts`, as well as to the order of variants specified in `pcbConfig`. To help with that, it makes sense to avoid early returns with different outcomes from convoluted loops such as this one. Your current code is not invariant: it will behave differently e.g. for `fit_field='+A,-B'` and `fit_field='-B,+A'`..

Reply:
Hi @RomaVis !
I already asked you about an example that should be a good start for test:

> Your example is here: [kibom-variant_2.sch](https://github.com/INTI-CMNB/KiBot/blob/master/tests/board_samples/kibom-variant_2.sch)
> Let me know if this is properly interpreted.

And I never got your opinion.

If you think the patch is not working please post a small example as the above, containing: a circuit and what we should get for each case. Then, with an example case, we can determine if the code does or doesn't what the docs say. Also: the example will be added to my regression tests to ensure the behavior doesn't change in the future.

Reply:
I may be able to prepare corresponding example schematic sometime next week (as I have no KiCad on my current system). Nevertheless, I thought that my comment already provides enough information for you to quickly check if the behavior of your patch is correct.

Reply:
Hi @RomaVis:
I adjusted the patch. This passes the following test case:  [kibom-variant_5.sch](https://github.com/INTI-CMNB/KiBot/blob/master/tests/board_samples/kibom-variant_5.sch)